src/GitVersionTask.MsBuild/GitVersionTask.MsBuild.csproj

```diff
@@ -1,7 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
     <PropertyGroup>
-        <TargetFrameworks>netstandard2.0</TargetFrameworks>
+        <TargetFrameworks>net472;netstandard2.0</TargetFrameworks>
```

Comment:
I think it was this that was the main part of the problem. This assembly needs to be cross compiled for full net framework and .net core, .net core msbuild ends up using the .net standard compiled one, and .net framework msbuild needs to use the full .net (net472) compiled version of the assembly - that's becuase in this assembly there is stuff like this:
```
  static TaskProxy()
        {
#if !NETFRAMEWORK
            GitLoaderContext.Init("GitVersionCore", "LibGit2Sharp");
#endif
            LibGit2SharpLoader.LoadAssembly("GitVersionTask");
```
If we only build the .net standard version of the binary to be used by both .net core and net framework msbuild variants, then it will now not include the full .NET dependency resolution logic. 

I had fixed this in my PR, but I broke other things when trying to refactor task proxy. I have decided to start my PR again from a clean branch off latest master, and I'll just do some more targetted fixes.

Reply:
I'm working now on adding some integration tests using msbuild and dotnet build to validate the GitVersionTask,  when that is in place I think we can release, and you can use that for validating the refactoring you're doing. As for the support for preview builds of msbuild I think we need to handle that after this release,